rafx-framework/src/visibility/visibility_object_allocator.rs

```diff
@@ -26,6 +26,19 @@ impl<'a> Deref for VisibilityObjectRef<'a> {
     }
 }
 
+pub struct VisibilityObjectLookup<'a>(
+    RwLockReadGuard<'a, SlotMap<VisibilityObjectId, VisibilityObject>>,
```

Comment:
How often would we be hitting this lock?

Reply:
Inside of `query_visibility` in the `ViewVisibilityJob`, every single `VisibilityResult` must be mapped back to a `VisibilityObject` to figure out what `RenderObject`s are associated with it. So this guard is created at the top of `query_visibility` and however many thousands of look-ups occur all resolve against the already acquired lock instead of needing a new lock each time.